optaplanner-quarkus-integration/optaplanner-quarkus/integration-test/src/main/java/org/optaplanner/quarkus/it/domain/ITestdataPlanningEntity.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.quarkus.it.domain;
+
+import org.optaplanner.core.api.domain.entity.PlanningEntity;
+import org.optaplanner.core.api.domain.variable.CustomShadowVariable;
+import org.optaplanner.core.api.domain.variable.PlanningVariable;
+import org.optaplanner.core.api.domain.variable.PlanningVariableReference;
+
+@PlanningEntity
+public class ITestdataPlanningEntity {
```

Comment:
Consider reusing stuff from `testdata` in `core`. You can include the test JAR into this Maven module.

Reply:
It does test a shadow variable to ensure custom variable listeners work (which is used in a constraint provider). Also, can cause issues since there are multiple `@PlanningSolution` classes (using a solverConfig the defines the solution class and entity classes might fix that). I think it be more work + code to reuse classes than creating new classes in this case.

Reply:
Ok, two options then:

- Put these to `testdata` in `core` and use them from there.
- Create a `testdata` package here, for consistency sake.

In either case, we do not use the `I` prefix. I understand you used it to say this testdata is for an integration test, but IMO that is not important.